cli/src/cmd/forge/script/mod.rs

```diff
@@ -410,6 +416,72 @@ impl ScriptArgs {
         };
         Ok((func.clone(), data))
     }
+
+    /// Checks if the transaction is a deployment with a size above the `CONTRACT_MAX_SIZE`.
+    ///
+    /// If `self.broadcast` is enabled, it asks confirmation of the user. Otherwise, it just warns
+    /// the user.
+    fn check_contract_sizes(
+        &self,
+        transactions: Option<&VecDeque<TypedTransaction>>,
+        known_contracts: &BTreeMap<ArtifactId, ContractBytecodeSome>,
+    ) -> eyre::Result<()> {
+        if let Some(txes) = transactions {
+            for tx in txes {
+                if let Some(data) = tx.data() {
+                    if data.len() > CONTRACT_MAX_SIZE {
+                        let mut offset = 0;
+
+                        // Find if it's a CREATE or CREATE2. Otherwise, skip transaction.
+                        if let Some(NameOrAddress::Address(to)) = tx.to() {
+                            if *to == DEFAULT_CREATE2_DEPLOYER {
+                                // Size of the salt prefix.
+                                offset = 32;
+                            }
+                        } else if tx.to().is_some() {
+                            continue
+                        }
```

Comment:
I'm not sure I fully understand here: does this only work for `CREATE2`?

The way I read it:

- If the deployer is the CREATE2 deployer, set `offset` to `32`
- Otherwise skip the transaction

Reply:
Ahhh, never mind - if `tx.to` is something and it's not the `create2` deployer then we're not deploying anything